src/txn/txn.c

```diff
@@ -837,6 +838,23 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
 		    session, "durable", ts, &cval, true));
 	}
 
+	/*
+	 * Confirm if the first commit timestamp is valid
+	 * with both prepared and unprepared transactions.
+	 */
+	if (txn_global->has_stable_timestamp &&
```

Comment:
I'm not sure if this is the best method to do this but it does avoid moving it into the assert function (which is presumably for when asserting is turned on) and it also avoids calling the much more expensive validate function.

Reply:
For prepared transactions, first commit timestamp will not be set. The condition '(prepare && !F_ISSET(txn, WT_TXN_HAS_TS_DURABLE))' is redundant.

Reply:
Better to move this check to '__wt_txn_commit_timestamp_validate'

Reply:
> For prepared transactions, first commit timestamp will not be set. The condition '(prepare && !F_ISSET(txn, WT_TXN_HAS_TS_DURABLE))' is redundant.

Does this mean that my test scenarios 4 and 5 are invalid? The test will fail if I remove this check, implying the first commit timestamp is set.